WordPress/gutenberg
# RichText: Add a format prop to allow HTML string values to be used in RichText components

We discussed several times whether we want to treat RichText values as strings, elements or raw trees. In this PR I'm introducing a `format` prop for the RichText component for two reasons:

 - If someone wants to store HTML in a meta attribute, the RichText component should produce a string output and not an array of element. This has been raised a while ago but I'm not able to find the corresponding issue.

 - We know we want to move away from the element's tree as default representation of RichText value (because it's not serializable properly on the server), we're not certain yet about the alternative. But anyway in order to do so (change the format) we need a migration plan, this `format` will allow us to introduce warnings and at some point switch the default `format` (which is `element` at the moment) and at a later point maybe remove the `element` format entirely.

**Testing instructions**

 - Right now, the newly introduce "string" format is not used anywhere, so this should not change anything for the existing blocks

 - You can setup a meta block with a "string" format for the RichText component.
